src/geom/cell_prism6.C

```diff
@@ -92,6 +92,14 @@ bool Prism6::is_node_on_side(const unsigned int n,
   return false;
 }
 
+std::vector<unsigned> Prism6::nodes_on_side(const unsigned int s) const
+{
+  libmesh_assert_less (s, n_sides());
+  return std::vector<unsigned>(side_nodes_map[s],
+                               side_nodes_map[s]
+                               + sizeof(side_nodes_map[s]) / sizeof(side_nodes_map[s][0]));
```

Comment:
Elements whose faces are different types, like `Prism6`, have dummy `99` entries in their `side_nodes_map`. I think we should definitely avoid returning those if possible, although this will complicate the implementation for those classes a bit.